src/test/java/com/example/name/service/NameServiceTest.java

```diff
@@ -51,4 +53,28 @@ public class NameServiceTest {
         });
         assertEquals("指定されたIDの名前は存在しません", exception.getMessage());
     }
+
+    @Test
+    public void 正常に新規のユーザーが登録できること() {
+        NameRequest newNameRequest = new NameRequest("kazushi", 33);
+        doNothing().when(nameMapper).insert(any(Name.class));
+        nameService.insert(newNameRequest);
+        verify(nameMapper).insert(any(Name.class));
```

Comment:
any()だと「どんな引数でも」ということになってしまいますので、
こうすることでより正確に引数を指定することができます！
```suggestion
        NameRequest newNameRequest = new NameRequest("kazushi", 33);
        doNothing().when(nameMapper).insert(newNameRequest);
        nameService.insert(newNameRequest);
        verify(nameMapper).insert(newNameRequest);
```
こうするには、entityのNameクラスでで、すでにやられているように、NameRequestで`equal`メソッドをオーバーライドしてあげないといけないですね！

Reply:
whenメソッドの引数に渡すオブジェクトを変更しました。柴田さんからの指摘でnewNameRequestを引き数に設定しようとしていましたが、Mapperクラスのinsertメソッドに NameRequest オブジェクトではなくName オブジェクトを渡すよう変更が必要だったのと、Serviceクラスにおいても修正が必要になり混乱してしまいました。
そのため、NameRequestのconvertToNameを引き数に設定し、equalメソッドをオーバーライドしたのですが適切でしょうか？あまり自信がないですが、一度確認して頂きたいです。

コミットハッシュ値